examples/triangle_intersection/triangle_intersection.cpp

```diff
@@ -16,358 +16,81 @@
-    }
-    return ArborX::CallbackTreeTraversalControl::normal_continuation;
-  }
-
-private:
-  Triangles<MemorySpace> _triangles;
-  Kokkos::View<int *, MemorySpace> _offsets;
-  Kokkos::View<ArborX::Point *, MemorySpace> _coefficients;
-};
-
-// Now that we have encapsulated the objects and queries to be used within the
-// Triangles class, we can continue with performing the actual search.
-int main()
-{
-  Kokkos::ScopeGuard guard;
+  Kokkos::ScopeGuard guard(argc, argv);
 
   using ExecutionSpace = Kokkos::DefaultExecutionSpace;
-  using MemorySpace = typename ExecutionSpace::memory_space;
-  ExecutionSpace execution_space;
-
-  // Create grid with triangles
-  Triangles<MemorySpace> triangles(execution_space);
-
-  // Create BVH tree
-  ArborX::BoundingVolumeHierarchy<MemorySpace,
-                                  ArborX::PairValueIndex<Box>> const
-      tree(execution_space, ArborX::Experimental::attach_indices(triangles));
-
-  // Create the points used for queries
-  Points<MemorySpace> points(execution_space);
-
-  // Execute the queries
-  int const n = points.size();
-  Kokkos::View<int *, MemorySpace> offsets("Example::offsets", n);
-  Kokkos::View<ArborX::Point *, MemorySpace> coefficients(
-      "Example::coefficients", n);
-
-  tree.query(execution_space, points,
-             TriangleIntersectionCallback<MemorySpace>{triangles, offsets,
-                                                       coefficients});
-
-  // Check the results
-  bool success;
-  Kokkos::parallel_reduce(
-      Kokkos::RangePolicy<ExecutionSpace>(execution_space, 0, n),
-      KOKKOS_LAMBDA(int i, bool &update) {
-        constexpr float eps = 1.e-3;
-
-#if KOKKOS_VERSION >= 40200
-        using Kokkos::printf;
-#elif defined(__SYCL_DEVICE_ONLY__)
-        using sycl::ext::oneapi::experimental::printf;
-#endif
-        if (offsets(i) != i)
-        {
-          printf("Offsets are wrong for query %d.\n", i);
-          update = false;
-        }
-        auto const &c = coefficients(i);
-        auto const &t = triangles(offsets(i));
-        auto const &p_ref = points(i);
-        Point p{c[0] * t.a[0] + c[1] * t.b[0] + c[2] * t.c[0],
-                c[0] * t.a[1] + c[1] * t.b[1] + c[2] * t.c[1]};
-        if ((Kokkos::abs(p[0] - p_ref[0]) > eps) ||
-            Kokkos::abs(p[1] - p_ref[1]) > eps)
-        {
-          printf("Coefficients are wrong for query %d.\n", i);
-          update = false;
-        }
-      },
-      Kokkos::LAnd<bool, Kokkos::HostSpace>(success));
-  std::cout << "Check " << (success ? "succeeded" : "failed") << '\n';
-  return success ? EXIT_SUCCESS : EXIT_FAILURE;
+  using MemorySpace = ExecutionSpace::memory_space;
+
+  using Point = ArborX::ExperimentalHyperGeometry::Point<2>;
+  using Triangle = ArborX::ExperimentalHyperGeometry::Triangle<2>;
+
+  //  6_____7_____8
+  //  |\  x |\   x|
+  //  |  \  |  \  |
+  //  3____\4____\5
+  //  |\    |\    |
+  //  |  \ x|x \  |
+  //  0____\1____\2
```

Comment:
It might be helpful to have 3 versions of your ASCII drawing,
* the 1st one showing the points supporting the triangles
* the 2nd one numbering the triangles
* the 3rd one showing the points associated to the queries